From 8b3c6f2905e8ad2b4a5f06afff0ee7e4c18b5c3a Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Thu, 16 Apr 2015 12:52:51 +0530
Subject: [PATCH] Whiskey Cove PMIC: Adding alert support for BAT0 to zone STR3

Issue description: Previously bat alerts were received in ccsm
driver and no processing was done for this interrupt. As a result
user space thermal management had to poll for the trip and take
actions.

Solution: This change adds the support for forwarding bat0 alerts
to user space for processing. This is done by registering the
interrupt to the pmic thermal resource and let the pmic thermal
driver take care of the posting the uevent to the user space.

Change-Id: I4b4e247732781242b3ed40655e14cccd096cefc2
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1444
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
---
 drivers/mfd/intel_soc_pmic_wc.c | 27 ++++++++++++++++++++++-----
 1 file changed, 22 insertions(+), 5 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_wc.c b/drivers/mfd/intel_soc_pmic_wc.c
index 79c3b59..b6b713d 100644
--- a/drivers/mfd/intel_soc_pmic_wc.c
+++ b/drivers/mfd/intel_soc_pmic_wc.c
@@ -301,11 +301,6 @@ static struct resource pmic_ccsm_resources[] = {
 		.flags = IORESOURCE_IRQ,
 	},
 	{
-		.start = BATALRT_IRQ,
-		.end   = BATALRT_IRQ,
-		.flags = IORESOURCE_IRQ,
-	},
-	{
 		.start = CTYPE_IRQ,
 		.end   = CTYPE_IRQ,
 		.flags = IORESOURCE_IRQ,
@@ -355,6 +350,11 @@ static struct resource thermal_resources[] = {
 		.end   = THRM3_IRQ,
 		.flags = IORESOURCE_IRQ,
 	},
+	{
+		.start = BATALRT_IRQ,
+		.end   = BATALRT_IRQ,
+		.flags = IORESOURCE_IRQ,
+	},
 };
 
 static struct resource bcu_resources[] = {
@@ -590,6 +590,18 @@ static struct trip_config_map str2_trip_config[] = {
 	},
 };
 
+static struct trip_config_map str3_trip_config[] = {
+	{
+		.irq_reg = THRM2IRQ,
+		.irq_mask = 0x01,
+		.irq_en = MTHRMIRQ2,
+		.irq_en_mask = 0x01,
+		.evt_stat = STHRMIRQ2,
+		.evt_mask = 0x10,
+		.trip_num = 1
+	},
+};
+
 static struct thermal_irq_map wc_thermal_irq_map[] = {
 	{
 		.handle = "STR0",
@@ -606,6 +618,11 @@ static struct thermal_irq_map wc_thermal_irq_map[] = {
 		.trip_config = str2_trip_config,
 		.num_trips = ARRAY_SIZE(str2_trip_config),
 	},
+	{
+		.handle = "STR3",
+		.trip_config = str3_trip_config,
+		.num_trips = ARRAY_SIZE(str3_trip_config),
+	},
 };
 
 static struct pmic_gpio_data whiskey_cove_gpio_data = {
-- 
1.9.1

